contracts/cw1-subkeys/src/contract.rs

```diff
@@ -193,11 +205,21 @@ where
         return Err(ContractError::CannotSetOwnAccount {});
     }
 
-    ALLOWANCES.update::<_, StdError>(deps.storage, &spender_addr, |allow| {
-        let mut allowance = allow.unwrap_or_default();
+    ALLOWANCES.update::<_, ContractError>(deps.storage, &spender_addr, |allow| {
+        let mut allowance = allow
+            .filter(|allow| !is_expired(&env, allow.expires))
```

Comment:
Good checks for expiration (and good error below).
I just request using the Expiration method